src/gui/widgets/TrackOperationsWidget.cpp

```diff
@@ -345,15 +354,15 @@ void TrackOperationsWidget::updateMenu()
 	}
 
 	toMenu->addSeparator();
-	toMenu->addAction( embed::getIconPixmap( "colorize" ),
-						tr( "Change color" ), this, SLOT( changeTrackColor() ) );
-	toMenu->addAction( embed::getIconPixmap( "colorize" ),
-						tr( "Reset color to default" ), this, SLOT( resetTrackColor() ) );
-	toMenu->addAction( embed::getIconPixmap( "colorize" ),
-						tr( "Set random color" ), this, SLOT( randomTrackColor() ) );
-	toMenu->addSeparator();
-	toMenu->addAction( embed::getIconPixmap( "colorize" ),
-						tr( "Clear clip colors" ), this, SLOT( useTrackColor() ) );
+
+	QMenu* colorMenu = new QMenu(tr("Track color"), this);
+	colorMenu->setIcon(embed::getIconPixmap("colorize"));
+	colorMenu->addAction(tr("Change"), this, SLOT(changeTrackColor()));
+	colorMenu->addAction(tr("Reset"), this, SLOT(resetTrackColor()));
+	colorMenu->addAction(tr("Pick random"), this, SLOT(randomTrackColor()));
+	colorMenu->addSeparator();
+	colorMenu->addAction(tr("Reset clip colors"), this, SLOT(resetTCOColors()));
+	toMenu->addMenu(colorMenu);
```

Comment:
Creating the menu like this means it will last until the `TrackOperationsWidget` itself is deleted, so menus will gradually accumulate in memory until the project is changed or the track deleted. Creating it using `QMenu::addMenu` means it will be a child of the other menu, not the `TrackOperationsWidget`, so it will be deleted along with that menu.
```suggestion
	QMenu* colorMenu = toMenu->addMenu(embed::getIconPixmap("colorize"), tr("Track color"));
	colorMenu->addAction(tr("Change"), this, SLOT(changeTrackColor()));
	colorMenu->addAction(tr("Reset"), this, SLOT(resetTrackColor()));
	colorMenu->addAction(tr("Pick random"), this, SLOT(randomTrackColor()));
	colorMenu->addSeparator();
	colorMenu->addAction(tr("Reset clip colors"), this, SLOT(resetTCOColors()));
```

Reply:
Phew good you found that!